### PR TITLE
ESLint: Use 'eslint-plugin-header' to autofix header comments

### PR DESCRIPTION
**Pull request by DLehenbauer**
This is redundant with the broader header check in 'repo-policy-check', but enables headers to conveniently be fixed via `npm run lint:fix`.

I usually need an extra trip through the PR gates to fix a missing header.  So, saves time for both me and Azure.  :-)

@curtisman - We could remove the redundancy w/'repo-policy-check' once 0.24 is deployed, if desired.  Let me know.

### CODE REVIEW

File: common/build/eslint-config-fluid/eslint7.js [hunk 1]
```diff
@@ -226,6 +227,20 @@ module.exports = {
...
+        //
+        // Note: While redundant with the broader header check in 'repo-policy-check', eslint enables early
+        //       detection during 'npm run build' and allows autofixing with both VS Code and 'npm run lint:fix'.
+        "header/header": [
+            "error",
+            "block",
+            {
+                pattern: /^!\n \* Copyright \(c\) Microsoft Corporation\. All rights reserved\.\n \* Licensed under the MIT License\.\n (\*\n \* THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY\n )?$/,
+                template: "!\n * Copyright (c) Microsoft Corporation. All rights reserved.\n * Licensed under the MIT License.\n "
+            },
```
Comment (DLehenbauer) [inline common/build/eslint-config-fluid/eslint7.js:240]: If we modify 'gen-version' to place the "THIS IS AN AUTOGENERATED FILE.." warning outside the header comment, we can simplify pattern/template to just a string:
```
        "header/header": [
            "error",
            "block",
            "!\n * Copyright (c) Microsoft Corporation. All rights reserved.\n * Licensed under the MIT License.\n ",
            /* newLines: */ 2, // '\n\n' = 1 blank line after block comment
        ],
```

**Comment by curtisman** [comment]
Let's keep the repo-policy-check, since it will make sure it will check whether eslint is running or not (some source files are not covered by eslint).

Although, maybe a better fix is just to have fluid-build run repo-policy-check?

**Comment by DLehenbauer** [comment]
Running a scoped repo-policy-check as part of `npm run lint` would be perfect and would diminish the returns on the eslint header plugin.

The only remaining convenience the eslint plugin offers is autofix support (via 'lint:fix' and inside VS code w/the eslint plugin.)

I don't feel strongly about it.  If no one advocates for the plugin in the next couple of days, I'll quietly close this PR.

**Comment by tylerbutler** [comment]
I don't have a strong opinion on this. In principle I prefer this approach to custom stuff in the fluid build tools. That said, since we have to keep the build tools around for file types not covered by eslint, I think we should stick with the build tools for now.

**Comment by DLehenbauer** [comment]
No advocates.  Closing PR.

**Pull request closed without merging**